pkg/sql/paramparse/paramobserver.go

```diff
@@ -195,6 +195,41 @@ var tableParams = map[string]tableParam{
 			return nil
 		},
 	},
+	`exclude_data_from_backup`: {
+		onSet: func(ctx context.Context, po *TableStorageParamObserver, semaCtx *tree.SemaContext,
+			evalCtx *tree.EvalContext, key string, datum tree.Datum) error {
+			if po.tableDesc.Temporary {
+				return pgerror.Newf(pgcode.FeatureNotSupported,
+					"cannot set data in a temporary table to be excluded from backup")
+			}
+
+			// Check that the table does not have any incoming FK references. During a
+			// backup, the rows of a table with ephemeral data will not be backed up, and
+			// could result in a violation of FK constraints on restore. To prevent this,
+			// we only allow a table with no incoming FK references to be marked as
+			// ephemeral.
+			if len(po.tableDesc.InboundFKs) != 0 {
+				return errors.New("cannot set data in a table with inbound foreign key constraints to be excluded from backup")
+			}
+
+			s, err := GetSingleBool(key, datum)
```

Comment:
do you want to allow `'on'` as well, as we do with outer variables?

if so, you may want to pull in https://github.com/cockroachdb/cockroach/commit/3bc72cfa9b2c46466f7a33086747ea84592271b3 and use `boolFromDatum`

Reply:
Hmm not sure  `exclude_data_from_backup = on` makes sense so leaving this as is for now

Reply:
it also allows `'true'` as a string, and is standard for session vars (e.g. `set experimental_enable_temp_tables = 'on'`)

Reply:
(although at the end of the day up to you)

Reply:
done! sprinkled 'on' and 'true' in the logictest too.